### ISSUE TITLE
Data provider Resource too Big

### ISSUE THREAD

**Issue by chrmor** [DM2E/dm2e-mappings#38]
I cannot consume resources like 

http://data.dm2e.eu/data/agent/ub-ffm/sammlungen/Universit%C3%A4tsbibliothek_JCS_Frankfurt_am_Main

They take ages to load as the contains a lot of triples.

However I need a way of getting at least the label of the content provider.
Would it be possible to put the relative triple in the representation of the CHO? so I do not have to follow this link?

What other alternatives?

**Comment by knepper** [comment]
There is just a single property in this resource:

``` xml
   <foaf:Organization rdf:about="http://data.dm2e.eu/data/agent/ub-ffm/sammlungen/Universit%C3%A4tsbibliothek_JCS_Frankfurt_am_Main">
      <skos:prefLabel xml:lang="de">Universitätsbibliothek JCS Frankfurt am Main</skos:prefLabel>
   </foaf:Organization>
```

Thats all.

But there are - automaticly generated -  links back to all resources using this resource. They might be the problem.

This would be a general problem because all CHO must have a link to the data provider. There is no workaround for the data mapping if this is the reason.

**Comment by kba** [comment]
Yes, this is a serious problem. The culprit here are the places, since very few of them relate to so many CHOs/Aggregations, i.e. Frankfurt pulls in all the data from UBFFM/Sammlungen, Berlin all the data from BBAW/DTA. I already tweaked the responsiple SPARQL query to only pull in book-level CHOs but it's still too slow. My workaround is to heavily use caching and get the places in the cache as early as possible and for as long as possible.

@kaiec 
If there is a way to drastically improve the query performance by reducing the recall, that would be really helpful. Or we could have a mechanism (like query parameters or an addition to the URL) that skips the querying for backlinks altogether.

If neither is possible, we'd need some form of de-normalization, e.g bundling even more data with the ResourceMap like the contextual classes with their skos:prefLabel and the TimeSpans (see #32). Data consumers can't be expected to wait 60 seconds to get the `skos:prefLabel` of a place.

**Comment by kba** [comment]
@chrmor As an ad-hoc workaround to get you going, I temporarily disabled the backlinking altogether until we find a solution to improve performance. http://data.dm2e.eu/data/html/agent/ub-ffm/sammlungen/Universit%C3%A4tsbibliothek_JCS_Frankfurt_am_Main should now load in a few ms.

@kaiec The title on http://data.dm2e.eu/data/html/agent/ub-ffm/sammlungen/Universit%C3%A4tsbibliothek_JCS_Frankfurt_am_Main is broken because of the umlaut but correctly displayed in the `skos:prefLabel`. Any idea why that is?

**Comment by chrmor** [comment]
Ok , then. But I do not understand one thing: will I find the same problem in other datasets? or you fixed it once of all?

**Comment by kba** [comment]
Yes, it is fixed for all contextual resources in all datasets. Drawback is, that you don't get backlinks for them anymore. But it should be very fast and easy now to get things like `skos:prefLabel` or `foaf:name` or whatever.

**Comment by chrmor** [comment]
Well this is a bit sad as we cannot use such backlinks e.g. for recommendations.
But this is a limit of the pure Linked Data interface. Fact is you would need SPARQL to implement such things.

**Comment by kba** [comment]
Yes, you're right and we do use SPARQL in the backend. The balancing act is to provide the interesting backlinks in a reasonable fast and general implementation without dragging down the performance and precision by pulling in every backlink in the store. We need to tweak the queries. Exposing the SPARQL endpoint is probably a bad idea because it shifts the responsibility of writing efficient queries to the data consumer -- and it is very easy to write inefficient queries in a store with hundreds of millions of triples and hundred thousands of graphs.

**Comment by chrmor** [comment]
Right.
Perhaps we need to implementare in the future some server sode logic to
filter "interesting backlinks". At least If we want "smart" behaviors in
clients like Feed.
Il 28/feb/2014 23:29 "kba" notifications@github.com ha scritto:

**Comment by kba** [comment]
As a short-term solution, maybe we can agree on a core set of specific properties and classes that you'd need links back from a contextual object, depending on what you actually need in Feed. You are probably not interested in all the aggregations that are related to a place (i.e. the fact that all the Aggregations by Wittgenstein are related to Bergen) but you are interested in the CHOs that are related to a place (i.e. all the CHOs that are on the subject of Bergen). Can we find a core set of such 'interesting' relations?

**Comment by knepper** [comment]
This solution is already working?

http://data.dm2e.eu/data/agent/ub-ffm/sammlungen/Universit%C3%A4tsbibliothek_JCS_Frankfurt_am_Main

has no backlinks while other useful backlinks are still there?

**Comment by knepper** [comment]
Seems to be solved - I close it.

**Comment by knepper** [comment]
Solved or not? Closed or open? Something to do for the mapping? :-)

**Comment by kba** [comment]
It's not really solved since there are no backlinks because having those backlinks makes frequently referenced resources like http://data.dm2e.eu/data/agent/ub-ffm/sammlungen/Universit%C3%A4tsbibliothek_JCS_Frankfurt_am_Main painfully slow.

Nothing to do for the data providers/mappers though.
